Formula/b/build2.rb

```diff
@@ -0,0 +1,149 @@
+class Build2 < Formula
+  desc "C/C++ Build Toolchain"
+  homepage "https://build2.org"
+  url "https://download.build2.org/0.16.0/build2-toolchain-0.16.0.tar.xz"
+  sha256 "23793f682a17b1d95c80bbd849244735ed59a3e27361529aa4865d2776ff8adc"
+  license "MIT"
+
+  # TODO: add livecheck based on https://download.build2.org/toolchain.sha256
+  #       for background, see https://download.build2.org/README
+
+  # Note that for now we use SQLite from Mac OS instead of Homebrew at the
+  # insistence of Homebrew maintainers but may switch in the future if have a
+  # good enough reason (for example, higher SQLITE_MAX_ATTACHED limit). See PR
+  # #141550 for details. Note also that on Linux it comes from Homebrew so
+  # we have to arrange for library search paths, etc (see below).
+  #
+  depends_on "pkgconf"
+
+  uses_from_macos "curl"
+  uses_from_macos "openssl"
+  uses_from_macos "sqlite"
+
+  def install
+    pkgconf = Formula["pkgconf"]
+    sqlite = Formula["sqlite"] if OS.linux?
+
+    # Note that the below steps are equivalent to invoking build.sh with the
+    # --local --system libpkgconf,libsqlite3 options. The only difference are
+    # a few customizations to find Homebrew's libpkgconf and libsqlite3.
+    #
+    # For background on these steps, see:
+    #
+    # https://build2.org/build2-toolchain/doc/build2-toolchain-install.xhtml
+
+    # Suppress loading of default options files.
+    #
+    ENV["BUILD2_DEF_OPT"] = "0"
+
+    # Bootstrap, stage 1.
+    #
+    # Note that we disable all warnings since we cannot do anything more
+    # granular during bootstrap stage 1.
+    #
+    cd "build2" do
```

Comment:
why not stay in the `build2` directory with `chdir`?

Reply:
Replaced all cases of `cd` with `chdir`, thanks for the tip!